backend/packages/Upgrade/src/api/services/QueryService.ts

```diff
@@ -34,39 +34,58 @@ export class QueryService {
         relations: ['metric', 'experiment'],
       })
     );
+
     const promiseResult = await Promise.all(promiseArray);
-    const analyzePromise = promiseResult.map((query) => this.logRepository.analysis(query));
-    const response = await Promise.allSettled(analyzePromise);
+    const analyzePromise = promiseResult.map((query) => {
+      if (query.experiment.type === EXPERIMENT_TYPE.FACTORIAL) {
```

Comment:
We should handle the case that query does not have experiment, since query could be undefined. (findOne returns undefined if nothing matches) Or, we should handle the case that there is not associated query earlier.

Reply:
Resolved this